endpoint/contracts/examples/treasury/MintTreasury.sol

```diff
@@ -24,17 +26,40 @@ contract MintTreasury {
         string calldata name,
         string calldata symbol,
         address account,
-        uint256 amount
+        uint256 amount,
+        uint256 chainId
     ) external onlyEndpoint {
-        address token = treasuryMapping[originalToken];
+        address token = treasuryMapping[chainId][originalToken];
         if (token == address(0)) {
             token = address(new TreasuryToken(name, symbol));
-            treasuryMapping[originalToken] = token;
+            treasuryMapping[chainId][originalToken] = token;
         }
         TreasuryToken(token).mint(account, amount);
     }
 
-    function burn(address token, uint256 amount) public {
+    function burn(
+        uint256 rid,
+        address rua,
+        address originalToken,
+        address token,
+        uint256 amount,
+        address recv
+    ) public {
+        require(
+            token == treasuryMapping[rid][originalToken],
+            "token not match"
+        );
         TreasuryToken(token).burnFrom(msg.sender, amount);
+        bytes memory data = abi.encodeWithSelector(
+            bytes4(keccak256("unlock(address,uint256,address)")),
```

Comment:
question just for clarity: which contract's function `unlock` this is calling?

Reply:
lockContract